Disable vector register GRA and reduction on 32bit x86

IBMJimmyk opened the pull request:
GRA does not properly assign xmm real registers to TR_VRF virtual
registers on 32bit x86. This can lead to code that produces an incorrect
result. GRA for vector registers is temporarily being disabled on 32bit
x86 until this issue is resolved.

The removal of GRA may cause a performance problem in autosimd reductions
so reductions have also been disabled. This was done by temporarily
disabling the use of getvelem in autosimd on 32bit x86 which is needed
for reduction.

Issue: #2035
Signed-off-by: jimmyk <jimmyk@ca.ibm.com>

0xdaryl commented on compiler/x/codegen/OMRCodeGenerator.cpp:
```diff
@@ -299,8 +299,15 @@ OMR::X86::CodeGenerator::initialize(TR::Compilation *comp)
-   self()->setLastGlobalVRF(self()->getLastGlobalFPR());
+   /*
+    * GRA does not work with vector registers on 32 bit due to a bug where xmm registers are not being assigned
+    * This will temporary disable GRA for vector registers on 32 bit until the problem has been resolved
```
* Please use periods at the end of your sentences rather than just line breaks.
* Rather than say this fix is temporary, please reference the issue you created to track the resolution of the problem in the comment instead.  Part of the work in that issue should be to re-enable this code and remove the comment.

0xdaryl commented on compiler/x/codegen/OMRCodeGenerator.cpp:
```diff
@@ -1027,11 +1034,21 @@ OMR::X86::CodeGenerator::getSupportsOpCodeForAutoSIMD(TR::ILOpCode opcode, TR::D
+       * GRA does not work with vector registers on 32 bit due to a bug where xmm registers are not being assigned
+       * This can potentially cause a performance problem in autosimd reductions
+       * This function is where AutoSIMD checks to see if getvelem is suppored for use in reductions
+       * getvelem is temporarily marked as unsupported until the GRA problem has been fixed.
```
* Please use periods at the end of your sentences.

* Rather than say this fix is temporary, please reference the issue you created to track the resolution of the problem in the comment instead.  Part of the work in that issue should be to re-enable this code and remove the comment.

0xdaryl commented:
Genie build is not required because an x86-only file was modified.

IBMJimmyk commented:
I fixed the concerns mentioned in the previous comments. I fixed the punctuation in the code comments and added references to Issue #2035 and the conditions for reenabling the code.
